packages/components/src/components/mobile-dialog/mobile-dialog.jsx

```diff
@@ -0,0 +1,94 @@
+import classNames        from 'classnames';
+import PropTypes         from 'prop-types';
+import React             from 'react';
+import ReactDOM          from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+import Icon              from '../icon';
+
+const MobileDialog = (props) => {
+    const { title, visible, children, container_el, wrapper_classname } = props;
+
+    const checkVisibility = () => {
+        if (props.visible) {
+            document.body.style.overflow = 'hidden';
+            document.getElementById(container_el).style.overflow = 'hidden';
+        } else {
+            document.body.style.overflow = null;
+            document.getElementById(container_el).style.overflow = null;
+        }
+    };
+
+    const scrollToElement = (parent, el) => {
+        const viewport_offset = el.getBoundingClientRect();
+        const hidden = viewport_offset.top + el.clientHeight + 20 > window.innerHeight;
+        if (hidden) {
+            const new_el_top = (window.innerHeight - el.clientHeight) / 2;
+            parent.scrollTop += viewport_offset.top - new_el_top;
+        }
+    };
+
+    // sometimes input is covered by virtual keyboard on mobile chrome, uc browser
+    const handleClick = (e) => {
+        if (e.target.tagName === 'INPUT' && e.target.type === 'number') {
+            const scrollToTarget = scrollToElement(e.currentTarget, e.target);
+            window.addEventListener('resize', scrollToTarget, false);
+
+            // remove listener, resize is not fired on iOS safari
+            window.setTimeout(() => {
+                window.removeEventListener('resize', scrollToTarget, false);
+            }, 2000);
+        }
+    };
```

Comment:
FYI, these functions are to handle a bug with the iOS keyboard resizing the container when it's triggered in the `MobileDialog` and also prevent scrolling of background elements like `body` or the `container` of `MobileDialog` when `MobileDialog` is rendered in the foreground.